include/gatekeeper_gk.h

```diff
@@ -24,13 +24,61 @@
 #include "gatekeeper_ipip.h"
 #include "gatekeeper_ggu.h"
 #include "gatekeeper_mailbox.h"
+#include "gatekeeper_lpm.h"
+
+/*
+ * The LPM supports 8-bit next hop,
+ * which at most has 128 different values.
+ */
+#define GK_MAX_NUM_POLICIES (128)
+
+#define GK_MAX_NUM_GRANTORS (4)
 
 /*
  * A flow entry can be in one of three states:
  * request, granted, or declined.
  */
 enum gk_flow_state { GK_REQUEST, GK_GRANTED, GK_DECLINED };
 
+enum gk_policy_action {
+	/* Forward the packet to a Grantor server. */
+	GK_FWD_GT,
+
+	/* Forward the packet to the back interface. */
+	GK_FWD_BCAK_NET,
+
+	/* Drop the packet. */
+	GK_DROP,
+};
+
+struct gk_policy {
+
+	enum gk_policy_action action;
+
+	/* Data that supports different policy actions. */
+	union {
+		int grantor_id;
```

Comment:
`struct flow_entry` can keep `grantor_id` as it is, all that need to happen is for those IDs to only point to `struct gk_policy`s whose action is `GK_FWD_GT`.

Another option is to change `grantor_id` in `struct flow_entry` to point directly to `struct gk_policy`. I don't know which approach is best at this point.